crates/astria-conductor/src/conductor.rs

```diff
@@ -10,39 +18,114 @@ use tokio::{
         signal,
         SignalKind,
     },
-    sync::watch,
-    task::JoinHandle,
-    time,
+    sync::{
+        mpsc,
+        oneshot,
+        watch,
+    },
+    task::{
+        spawn_local,
+        LocalSet,
+    },
+    time::timeout,
 };
+use tokio_util::task::JoinMap;
 use tracing::{
     error,
     info,
+    warn,
 };
 
 use crate::{
-    driver::DriverCommand,
+    block_verifier::BlockVerifier,
+    client_provider::ClientProvider,
+    data_availability,
+    executor::Executor,
+    sequencer,
     Config,
-    Driver,
 };
+
 pub struct Conductor {
     signals: SignalReceiver,
-    driver: Driver,
-    executor_join_handle: JoinHandle<eyre::Result<()>>,
-    reader_join_handle: Option<JoinHandle<eyre::Result<()>>>,
+    /// the different long-running tasks that make up the conductor;
+    tasks: JoinMap<&'static str, eyre::Result<()>>,
+    /// channels to the long-running tasks to shut them down gracefully
+    shutdown_channels: HashMap<&'static str, oneshot::Sender<()>>,
+    sequencer_client_pool: deadpool::managed::Pool<ClientProvider>,
 }
 
 impl Conductor {
+    const DATA_AVAILABILITY: &str = "data_availability";
+    const EXECUTOR: &str = "executor";
+    const SEQUENCER: &str = "sequencer";
```

Comment:
Not in this case because I only want these as keys into the `JoinMap` and for logging events. Using an enum would would just make this more complex (and either require implementin/deriving a bunch of traits or always going `enum variant -> str`).